cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -161,36 +163,64 @@ public EnvironmentVariables resolved() {
   @Override
   public String resolve(String string, Object src) {
 
-    return resolve(string, src, 0, src, string);
+    return resolve(string, src, 0, src, string, this);
   }
 
-  private String resolve(String value, Object src, int recursion, Object rootSrc, String rootValue) {
+  private String resolve(String value, Object src, int recursion, Object rootSrc, String rootValue,
+      AbstractEnvironmentVariables resolvedVars) {
```

Comment:
`resolvedVars` is passed as `this` but never changed. Recursive invocations also always pass the same instance so if I am not mistaken, this parameter is pointless.

Reply:
I added javadoc to explain how resolvedVars is used and why it is necessary.